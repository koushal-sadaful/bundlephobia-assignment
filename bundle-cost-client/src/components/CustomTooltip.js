import React from "react";
import {prettifySize} from "../utils/sizeHelper";
import {Message} from "semantic-ui-react";

const CustomTooltip = (props) => {

    const {active} = props;
    if (active) {
        const {payload, label} = props;
        if (payload && payload.length > 0)
            return (
                <Message floating>
                    <p className="tooltip-label">{`version: ${label}`}</p>
                    <p className="tooltip-label">{`${payload[0].dataKey}: ${prettifySize(payload[0].value)}`}</p>
                    <p className="tooltip-label">{`${payload[1].dataKey}: ${prettifySize(payload[1].value)}`}</p>
                </Message>
            );
    }
    return (<div/>)
};

export default CustomTooltip