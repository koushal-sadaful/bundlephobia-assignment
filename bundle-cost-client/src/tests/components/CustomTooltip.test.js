import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, mount} from 'enzyme';
import {Message} from "semantic-ui-react";
import CustomTooltip from "../../components/CustomTooltip";
import {runTheSnapshotTest} from "../testUtils/TestHelpers";

configure({adapter: new Adapter()});

describe("CustomTooltip", () => {
    let wrapper = null;
    let fakePayload = [
        {
            dataKey: "gzip",
            name: "gzip",
            value: 4225,
            payload: {
                name: "0.18.1",
                minified: 11357,
                gzip: 4225,
            }
        }, {
            dataKey: "minified",
            name: "minified",
            value: 11357,
            payload: {
                name: "0.18.1",
                minified: 11357,
                gzip: 4225,
            }
        }
    ];

    let defaultTestProps = {
        active: true,
        label: "0.18.1",
        payload: fakePayload
    };

    afterEach(() => {
        if (wrapper){
            wrapper.unmount()
            wrapper = null
        }
    });

    it("should not render anything if is not active", () => {
        const testProps = {...defaultTestProps, ...{active: false}};
        wrapper = mount(<CustomTooltip {...testProps} />);
        expect(wrapper.find(Message).exists()).toEqual(false);
    });

    it("should render the label and sizes in the tooltip", () => {
        wrapper = mount(<CustomTooltip {...defaultTestProps} />);
        const tooltipText = wrapper.find(Message).props().children.map(child => child.props.children);
        expect(tooltipText).toEqual(['version: 0.18.1', 'gzip: 4.22 kB', 'minified: 11.4 kB']);
    });

    it("should not render anything if missing data", () => {
        const testProps = {...defaultTestProps, ...{payload: []}};
        wrapper = mount(<CustomTooltip {...testProps} />);
        expect(wrapper.find(Message).exists()).toEqual(false);
    });

    it("should pass the snapshot test", () => {
        runTheSnapshotTest(<CustomTooltip {...defaultTestProps} />)
    });

});