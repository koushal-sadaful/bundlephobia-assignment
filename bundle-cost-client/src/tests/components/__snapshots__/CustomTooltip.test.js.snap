// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip should pass the snapshot test 1`] = `
<div
  className="ui floating message"
>
  <p
    className="tooltip-label"
  >
    version: 0.18.1
  </p>
  <p
    className="tooltip-label"
  >
    gzip: 4.22 kB
  </p>
  <p
    className="tooltip-label"
  >
    minified: 11.4 kB
  </p>
</div>
`;
